databricks/koalas/groupby.py

```diff
@@ -1401,6 +1401,70 @@ def ffill(self, limit=None):
 
     pad = ffill
 
+    def head(self, n=5):
+        """
+        Return first n rows of each group.
+
+        Returns
+        -------
+        DataFrame or Series
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 1, 1, 1, 2, 2, 2, 3, 3, 3],
+        ...                    'b': [2, 3, 1, 4, 6, 9, 8, 10, 7, 5],
+        ...                    'c': [3, 5, 2, 5, 1, 2, 6, 4, 3, 6]},
+        ...                   columns=['a', 'b', 'c'],
+        ...                   index=[7, 2, 4, 1, 3, 4, 9, 10, 5, 6])
+        >>> df
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        4   1   1  2
+        1   1   4  5
+        3   2   6  1
+        4   2   9  2
+        9   2   8  6
+        10  3  10  4
+        5   3   7  3
+        6   3   5  6
+
+        >>> df.groupby('a').head(2)
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        10  3  10  4
+        5   3   7  3
+        3   2   6  1
+        4   2   9  2
+
+        >>> df.groupby('a')['b'].head(2)
+        7      2
+        2      3
+        10    10
+        5      7
+        3      6
+        4      9
+        Name: b, dtype: int64
+        """
+        groupkeys = self._groupkeys
+        sdf = self._kdf._sdf
+        window = Window.partitionBy([s._scol for s in groupkeys]) \
+                       .orderBy(F.monotonically_increasing_id())
+        sdf = sdf.select("*", F.when(F.row_number().over(window) <= n, True)).dropna()
+
+        if isinstance(self, DataFrameGroupBy):
+            internal = self._kdf._internal.copy(
+                sdf=sdf,
+                column_scols=[scol_for(sdf, col) for col in self._kdf._internal.data_columns])
+        else:
+            internal = self._kdf._internal.copy(
+                sdf=sdf,
+                column_index=[(self._kser.name,)],
+                column_scols=[scol_for(sdf, self._kser.name)])
```

Comment:
`self._kser._internal.data_columns[0]` instead of `self._kser.name`?
Those are different concepts.
The one is internal metadata, the other is external data. Please distinguish them.

Reply:
Thanks for the comment!
although they have same value, we should distinguish them to keep concept.
am i understood right?